.github/actions/update-config/action.yaml

```
@@ -48,58 +48,56 @@ runs:
         git config --local user.name "Interop Labs CI"
         git checkout -b ${{ steps.branch-name.outputs.branch }}
 
-        IFS=',' read -ra NETWORK_LIST <<< "${{ inputs.networks }}"
-        for NETWORK in "${NETWORK_LIST[@]}"; do
-          echo "Generating config for $NETWORK"
-          poetry run python generate.py --network "$NETWORK" gen-configs all-configs
-        done
-
+        NETWORK=${{ inputs.network }}
+        echo "Generating config for $NETWORK"
+        poetry run python generate.py --network "$NETWORK" gen-configs all-configs
         git add --all
-        git commit -m "chore: update generated configuration files for ${{ inputs.networks }}"
+        git commit -m "chore: update generated configuration files for ${{ inputs.network }}"
         git push --set-upstream origin ${{ steps.branch-name.outputs.branch }}
 
     - name: Create Pull Request
       shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.cicd_token }}
       id: create_pr
       run: |
         cd config-repo
-        PR_TITLE="chore: update configs for ${{ inputs.networks }} from ${{ inputs.source_repo }}"
+        PR_TITLE="chore: update configs for ${{ inputs.network }} from ${{ inputs.source_repo }}"
         PR_BODY="Auto-generated PR with updated configuration files"
         PR_HEAD="${{ steps.branch-name.outputs.branch }}"
         PR_BASE="main"
-        # Capture both response and HTTP status code
-        RESPONSE=$(curl -s -w "%{http_code}" -o response.json -X POST \
-          -H "Authorization: token ${{ inputs.cicd_token }}" \
-          -H "Accept: application/vnd.github.v3+json" \
-          https://api.github.com/repos/${{ inputs.config_repo }}/pulls \
-          -d "{\"title\":\"$PR_TITLE\",\"body\":\"$PR_BODY\",\"head\":\"$PR_HEAD\",\"base\":\"$PR_BASE\"}")
-        HTTP_STATUS=$(echo -n "$RESPONSE" | tail -c 3)
-        cat response.json
-        PR_NUMBER=$(jq -r .number response.json)
-        if [[ "$HTTP_STATUS" != "201" ]] || [[ "$PR_NUMBER" == "null" || -z "$PR_NUMBER" ]]; then
-          echo "Failed to create PR. HTTP status: $HTTP_STATUS, PR number: $PR_NUMBER"
-          cat response.json
-          exit 1
+
+        echo "Creating PR: $PR_TITLE"
+        CREATE_PR_URL=$(gh pr create \
+          --repo "${{ inputs.config_repo }}" \
+          --title "$PR_TITLE" \
+          --body "$PR_BODY" \
+          --head "$PR_HEAD" \
+          --base "$PR_BASE")
+        CREATE_PR_EXIT_CODE=$?
+
+        if [ "$CREATE_PR_EXIT_CODE" -ne 0 ]; then
+            echo "Failed to create PR."
+            exit 1
         fi
+        PR_NUMBER=$(gh pr view "$CREATE_PR_URL" --repo "${{ inputs.config_repo }}" --json number -q .number)
```

Was `jq` supposed to be used here somehow?
```suggestion
        PR_NUMBER=$(gh pr view "$CREATE_PR_URL" --repo "${{ inputs.config_repo }}" --json | jq -q .number)
```

---

@milapsheth no, the command line has a built-in jq expression filtering